src/llmq/quorums_chainlocks.h

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2019 The Dash Core developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef DASH_QUORUMS_CHAINLOCKS_H
+#define DASH_QUORUMS_CHAINLOCKS_H
+
+#include "llmq/quorums.h"
+#include "llmq/quorums_signing.h"
+
+#include "net.h"
+#include "chainparams.h"
+
+class CBlockIndex;
+class CScheduler;
+
+namespace llmq
+{
+
+class CChainLockSig
+{
+public:
+    int32_t nHeight{-1};
+    uint256 blockHash;
+    CBLSSignature sig;
+
+public:
+    ADD_SERIALIZE_METHODS
+
+    template<typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
+    {
+        READWRITE(nHeight);
+        READWRITE(blockHash);
+        READWRITE(sig);
+    }
+
+    std::string ToString() const;
+};
+
+class CChainLocksHandler : public CRecoveredSigsListener
+{
+    static const int64_t CLEANUP_INTERVAL = 1000 * 30;
+    static const int64_t CLEANUP_SEEN_TIMEOUT = 5 * 60 * 1000;
```

Comment:
Isn't dropping "seen" after 5 minutes a bit too aggressive and defeats the purpose of having `seenChainLocks` though? It can take 10-20 minutes to mine a block sometimes.

Reply:
Hmm, true :) Changed to 1 day.